doc/setup-commands.rst

```diff
@@ -1100,6 +1102,10 @@ Copy the files into the install locations and (for library packages)
 register the package with the compiler, i.e. make the modules it
 contains available to programs.
 
+Additionally for GHC the ``extra-compilation-artifacts`` directory is copied if present.
+GHC plugins can store extra data in subfolders.
+(e.g. *extra-compilation-artifacts/PLUGIN_NAME/HS_MODULE.txt*)
```

Comment:
I don't believe this format is required. We copy _everything_ under `extra-compliation-artifacts`. 

> It is advisable--though, not required--that plugins choose an identifier for themselves under `extra-compilation-artifacts` to prevent clashing with other plugins. As far as cabal is concerned it treats `extra-compilation-artifacts` as a black box (e.g. a plugin could put arbitrary files under *extra-compliation-artifcats/PLUGIN_ID/*).